bundles/net.tourbook.cloud/src/net/tourbook/cloud/strava/PrefPageStrava.java

```diff
@@ -266,11 +266,13 @@ private void createUI_30_TourUpload(final Composite parent) {
             GridDataFactory.fillDefaults().applyTo(_chkUseTourTypeMapping);
             _chkUseTourTypeMapping.setText(Messages.PrefPage_UploadConfiguration_Button_UseTourTypeMapping);
             _chkUseTourTypeMapping.setToolTipText(Messages.PrefPage_UploadConfiguration_Button_UseTourTypeMapping_Tooltip);
+            _chkUseTourTypeMapping.addSelectionListener(widgetSelectedAdapter(
+                  selectionEvent -> _linkTourTypeFilters.getControl().setEnabled(_chkUseTourTypeMapping.getSelection())));
          }
```

Comment:
The enable/disable method should be called not individually on each control. Mostly the `enable...()` method is used even when there are controls enabled/disabled which are not currently involved in the current action. 

When each control would be individually enabled/disabled in its own selection event, it would be a mess when there are many controls. This philosophy is implemented all over MT

![image](https://user-images.githubusercontent.com/1283445/139523727-832a7656-9a97-442a-9897-503dded9c7be.png)

Reply:
When testing the newly created Strava tour type filters without adding tour types to the filter, so the filter was empty, there were several issues in the old MT code which I've fixed with https://github.com/wolfgang-ch/mytourbook/commit/baf99ddd8bd5499abf9be342fd8a9d84df2d6011

Reply:
10+ year old!!

Reply:
> The enable/disable method should be called not individually on each control. Mostly the `enable...()` method is used even when there are controls enabled/disabled which are not currently involved in the current action.
> 
> When each control would be individually enabled/disabled in its own selection event, it would be a mess when there are many controls. This philosophy is implemented all over MT
> 

I usually call `enableControls()` in my code but because the Strava preference page is small and only 1 label needed to be enabled, I called it individually. Fixed in my [PR](https://github.com/wolfgang-ch/mytourbook/pull/415)

Reply:
> 
> 
> 10+ year old!!

Created 2007-10-27 488436d17eaa4a7ece3c50519e0c87cb77bc1706